functests/1_performance/performance.go

```diff
@@ -189,6 +194,83 @@ var _ = Describe("[rfe_id:27368][performance]", func() {
+					},
+					NodeSelector: map[string]string{newLabel: ""},
+					RealTimeKernel: &performancev1alpha1.RealTimeKernel{
+						Enabled: pointer.BoolPtr(true),
+					},
+					AdditionalKernelArgs: []string{
+						"NEW_ARGUMENT",
+					},
+					NUMA: &performancev1alpha1.NUMA{
+						TopologyPolicy: pointer.StringPtr("restricted"),
+					},
+				},
+			}
+			Expect(testclient.Client.Create(context.TODO(), secondProfile)).ToNot(HaveOccurred())
+
+			By("Checking that new KubeletConfig and MachineConfig created")
+			key := types.NamespacedName{
+				Name:      components.GetComponentName(secondProfile.Name, components.ComponentNamePrefix),
+				Namespace: components.NamespaceNodeTuningOperator,
+			}
+			kubeletConfig := &machineconfigv1.KubeletConfig{}
+			err := testclient.GetWithRetry(context.TODO(), key, kubeletConfig)
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("cannot find KubeletConfig object %s", key.Name))
+			Expect(kubeletConfig.Spec.MachineConfigPoolSelector.MatchLabels[machineconfigv1.MachineConfigRoleLabelKey]).Should(Equal(newRole))
+			Expect(kubeletConfig.Spec.KubeletConfig.Raw).Should(ContainSubstring("restricted"))
+
+			machineConfig := &machineconfigv1.MachineConfig{}
+			err = testclient.GetWithRetry(context.TODO(), key, machineConfig)
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("cannot find MachineConfig object %s", key.Name))
+			Expect(machineConfig.Labels[machineconfigv1.MachineConfigRoleLabelKey]).Should(Equal(newRole))
+			Expect(machineConfig.Spec.KernelArguments).Should(ContainElement("NEW_ARGUMENT"))
+
+			By("Checking that the initial MCP does not start updating")
+			Consistently(func() corev1.ConditionStatus {
+				return mcps.GetConditionStatus(testutils.RoleWorkerCNF, machineconfigv1.MachineConfigPoolUpdating)
+			}, 30, 5).Should(Equal(corev1.ConditionFalse))
+
+			By("Checking that tuned nodes do not have arguments from new profile")
+			for _, node := range workerRTNodes {
+				Expect(nodes.ExecCommandOnNode([]string{"cat", "/proc/cmdline"}, &node)).NotTo(ContainSubstring("NEW_ARGUMENT"))
+				Expect(nodes.ExecCommandOnNode([]string{"cat", "/rootfs/etc/kubernetes/kubelet.conf"}, &node)).NotTo(ContainSubstring("restricted"))
+			}
+
+			By("Remove second profile and verify that initial KubeletConfig and MachineConfig still exist")
+			Expect(testclient.Client.Delete(context.TODO(), secondProfile)).ToNot(HaveOccurred())
+			key = types.NamespacedName{
+				Name:      components.GetComponentName(profile.Name, components.ComponentNamePrefix),
+				Namespace: components.NamespaceNodeTuningOperator,
+			}
+			err = testclient.GetWithRetry(context.TODO(), key, &machineconfigv1.KubeletConfig{})
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("cannot find KubeletConfig object %s", key.Name))
+			err = testclient.GetWithRetry(context.TODO(), key, &machineconfigv1.MachineConfig{})
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("cannot find MachineConfig object %s", key.Name))
+
```

Comment:
Should we check again that the initial MCP does not start updating during delete?

Reply:
I'm not sure we need to cover it since it is the MCO logic, not PAO..  but yeah, maybe it would be good to add one more verification just in case